File: guides/micronaut-oracle-cloud-streaming/chess-game/java/src/main/java/example/micronaut/chess/dto/Player.java
package example.micronaut.chess.dto;

import com.fasterxml.jackson.annotation.JsonValue;

public enum Player {
    WHITE("w"),
    BLACK("b");

    private final String color;

    Player(String color) {
        this.color = color;
    }

    @JsonValue
    public String toString() {
        return this.color;
    }
}

File: guides/micronaut-oracle-cloud-streaming/chess-listener/python/src/example/micronaut/chess/dto/player.py
from enum import Enum

from com.fasterxml.jackson.annotation import JsonValue
from micronaut.serde.annotation import Serdeable


@Serdeable
class Player(Enum):
    WHITE = "w"
    BLACK = "b"

    @JsonValue
    def to_string(self) -> str:
        return self.value
